lib/logstasher/device/udp.rb

```diff
@@ -0,0 +1,40 @@
+# simple UDP logger
+
+require 'logstasher/device'
+require 'socket'
+
+
+module LogStasher
+  module Device
+    class UDP
+      include ::LogStasher::Device
+
+      attr_reader :options, :socket
+
+      def initialize(options = {})
+        @options = default_options.merge(stringify_keys(options))
+        @socket = UDPSocket.new
+      end
+
+      def close
+        @socket.close
+      end
+
+      def write(log)
+        @socket.send(log, 0, options['hostname'], options['port'])
```

Comment:
We should probably check the number of bytes sent to ensure the whole message was written. I'm sure there are helpers out there for this.

Reply:
what should be done if this happens? what would be the idiomatic way to warn of this inside a gem?

i was able to find something that spoke syslog udp, but i think we dont need to speak "full" syslog UDP since all of the information we need is already inside the json payload, so we can use the standard udp listener on filebeat/logstash.

Reply:
```Warning.warn ?```

Reply:
You should always be able to send UDP, so I'd say this should be fatal.

Reply:
(if you don't want to retry packets...)